File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        row = [0] * m
        col = [0] * n
        
        for i in range(m):
            for j in range(n):
                if mat[i][j] == 1:
                    row[i] += 1
                    col[j] += 1
        
        result = 0
        
        for i in range(m):
            if row[i] > 1:
                continue
                
            for j in range(n):
                if mat[i][j] == 1:
                    if row[i] == 1 and col[j] == 1:
                        result += 1
                    break
                    
        return result

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        return bruteForce(mat);
    }
private:
    // O(mn * (m + n)) Time and O(1) extra space
    int bruteForce(vector<vector<int>>& mat) {
        int m = mat.size(), n = mat[0].size();
        int result = 0;
        
        for(int i = 0; i < m; ++i) {
            for(int j = 0; j < n; ++j) {
                if(mat[i][j] == 1) {
                    int row = 0;
                    while(row < m) {
                        if(row != i && mat[row][j] == 1)
                            break;
                        ++row;
                    }
                    int col = 0;
                    while(col < n) {
                        if(col != j && mat[i][col] == 1)
                            break;
                        ++col;
                    }
                    if(row == m && col == n) ++result;
                }
            }
        }
        return result;
    }
};